File: 2018/solutions/prob03.c
#include <stdio.h>

int main(int argc, char **argv)
{
    double srv_hours, srv_rate, stor_gb, stor_rate, bw, bw_rate;
    double srv_cost, stor_cost, bw_cost, total_cost;

    while (scanf("%lf %lf %lf %lf %lf %lf", &srv_hours, &srv_rate,
                 &stor_gb, &stor_rate, &bw, &bw_rate) == 6) {
        srv_cost = srv_hours * srv_rate;
        stor_cost = stor_gb * stor_rate;
        bw_cost = bw * bw_rate;

        total_cost = srv_cost + stor_cost + bw_cost;

        printf("%.2f\n", total_cost);
    }
    return 0;
}

File: 2018/solutions/prob03.java
import java.io.*;

public class prob03 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader
                = new BufferedReader(new InputStreamReader(System.in));

        String line = null;
        while ((line = reader.readLine()) != null) {
            String[] data = line.split(" ");
            double srv_hours = Double.parseDouble(data[0]);
            double srv_rate = Double.parseDouble(data[1]);
            double stor_gb = Double.parseDouble(data[2]);
            double stor_rate = Double.parseDouble(data[3]);
            double bw_gb = Double.parseDouble(data[4]);
            double bw_rate = Double.parseDouble(data[5]);

            double srv_cost = srv_hours * srv_rate;
            double stor_cost = stor_gb * stor_rate;
            double bw_cost = bw_gb * bw_rate;
            double total_cost = srv_cost + stor_cost + bw_cost;
            total_cost = Math.round(total_cost * 100.0) / 100.0;

            System.out.printf("%.2f\n", total_cost);
        }
    }
}
